agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -192,16 +192,21 @@ func TestSquidProxy(t *testing.T) {
 	squidContainerJSON, err := client.ContainerInspect(ctx, squidContainer.ID)
 	require.NoError(t, err)
 
+	squidIP := ""
+	if squidContainerJSON.NetworkSettings != nil {
+		squidIP = squidContainerJSON.NetworkSettings.IPAddress
+	}
+	require.NotEmpty(t, squidIP, "Need to have squid proxy container's ip but it's empty")
+
 	// Squid startup time
 	time.Sleep(1 * time.Second)
-	t.Logf("Started squid container: %v", squidContainerJSON.Name)
+	t.Logf("Started squid container: %s", squidContainerJSON.Name)
 
 	agent := RunAgent(t, &AgentOptions{
 		ExtraEnvironment: map[string]string{
-			"HTTP_PROXY": "squid:3128",
+			"HTTP_PROXY": fmt.Sprintf("http://%s:3128", squidIP),
 			"NO_PROXY":   "169.254.169.254,/var/run/docker.sock",
 		},
-		ContainerLinks: []string{squidContainerJSON.Name + ":squid"},
```

Comment:
not totally clear on why this is going away.  What's the `ContainerLinks` field doing and why can we get rid of it?  (...same question applies below where this is used in the hostConfig.Links.)

Reply:
Container link enables containers launched in bridge mode to discover each other without ip (see [here](https://docs.docker.com/network/links/#communication-across-links)). So when the agent starts with bridge mode and specify squid container as the link, it can set HTTP_PROXY to "squid_container_name:port" without the need to know the squid container's ip (Docker will resolve that). Since we now want to start agent in host mode, we can't use that anymore (container link is only supported for bridge mode), and instead we get the squid container's ip from container inspect result and use it in HTTP_PROXY

Reply:
As far as I understand, containerLink made it possible to set HTTP_PROXY using _squid_ word, as the name of the proxy squid container is _squid_, instead we can just pass the ip address on which squid proxy is listening to.